tests/tests/gpos_tests.cpp

```diff
@@ -640,18 +641,23 @@ BOOST_AUTO_TEST_CASE( voting )
       BOOST_CHECK_EQUAL(db.get_global_properties().parameters.gpos_period_start(), now.sec_since_epoch());
 
       advance_x_maint(5);
-      // a new GPOS period is in but vote from user is before the start so his voting power is 0
+      // a new GPOS period is in but vote from user is before the start. WHo ever votes in 6th sub-period, votes will carry
       now = db.head_block_time();
       BOOST_CHECK_EQUAL(db.get_global_properties().parameters.gpos_period_start(), now.sec_since_epoch());
 
       generate_block();
 
+      // we are in the second GPOS period, at subperiod 1,
       witness1 = witness_id_type(1)(db);
       witness2 = witness_id_type(2)(db);
       BOOST_CHECK_EQUAL(witness1.total_votes, 0);
-      BOOST_CHECK_EQUAL(witness2.total_votes, 0);
+      //It's critical here, since bob votes in 6th sub-period of last vesting period, witness2 should retain his votes
```

Comment:
For clarity. Can you add some comment in the place where bob is voting in the 6th period ? I cant find where this is happening. Thanks.

Reply:
added for all sub-periods